tables/idxutils.py

```diff
@@ -471,7 +471,7 @@ def nextafter(x, direction, dtype, itemsize):
     direction."""
 
     assert direction in [-1, 0, +1]
-    assert dtype.kind == "S" or type(x) in (bool, int, float)
+    assert dtype.kind == "S" or type(x) in (bool, int, long, float)
```

Comment:
But this is just an additional precondition in entterning the `nextafter()` function, so not sure why this is helping.  Also, this crashes on Python 3 with "NameError: name 'long' is not defined".  Could you provide a patch that supports Python 3 too?